object_tracker/labels.go

```diff
@@ -36,19 +36,19 @@ func (t *myTracker) RenameFromMatches(matches []int, oldDets, newDets []objdet.D
 	// Go through valid matches and update name and track
 	for oldIdx, newIdx := range matches {
 		if newIdx != -1 {
-			newDets[newIdx] = ReplaceLabel(newDets[newIdx], oldDets[oldIdx].Label())
-			t.UpdateTrack(newDets[newIdx])
-			delete(notUsed, newIdx)
+			if matchinMtx[oldIdx][newIdx] != 0 {
+				newDets[newIdx] = ReplaceLabel(newDets[newIdx], oldDets[oldIdx].Label())
```

Comment:
same with accessing newDets and oldDets -- it would be safer to get the lengths of these slices and then do a quick comparison to make sure the indices fall within the lengths

Reply:
done